<?xml version="1.0"?>

<launch>
  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen" respawn="true">

    <!-- Frame names -->
    <param name="map_frame" value="map" />
    <param name="base_frame" value="base_range_sensor_link" />
    <param name="odom_frame" value="nav" />

    <!-- Tf use -->
    <param name="use_tf_scan_transformation" value="false"/>
    <param name="use_tf_pose_start_estimate" value="false"/>
    <param name="pub_map_odom_transform" value="false"/>

    <!-- Map size / start point -->
    <param name="map_resolution" value="0.050"/>
    <param name="map_size" value="2048"/>
    <param name="map_start_x" value="0.5"/>
    <param name="map_start_y" value="0.5" />
    <param name="map_multi_res_levels" value="2" />
    <remap from="map" to="/update_map"/>

    <!-- Map update parameters -->
    <param name="update_factor_free" value="0.39"/>
    <param name="update_factor_occupied" value="0.85" />
    <param name="map_update_distance_thresh" value="0.10"/>
    <param name="map_update_angle_thresh" value="2.0" />
    <param name="laser_max_dist" value="1.0" />
    <param name="laser_z_min_value" value="-1.0" />
    <param name="laser_z_max_value" value="1.0" />

    <!-- Advertising config -->
    <param name="advertise_map_service" value="true"/>

    <param name="scan_subscriber_queue_size" value="1"/>
    <param name="scan_topic" value="/hsrb/base_scan"/>

    <!-- Debug parameters -->
    <!--
      <param name="output_timing" value="false"/>
      <param name="pub_drawings" value="true"/>
      <param name="pub_debug_output" value="true"/>
    -->
    <param name="tf_map_scanmatch_transform_frame_name" value="scanmatcher_frame" />
  </node>

<!--  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find em_active_slam)/launch/hsr_active.rviz"/> -->

  <!--<node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster" args="0 0 0 0 0 0 map nav 100"/>-->
</launch>
